src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -863,16 +863,22 @@ static void
 EnsureLocalTableEmptyIfNecessary(Oid relationId, char distributionMethod,
 								 bool viaDepracatedAPI)
 {
-	if (viaDepracatedAPI)
-	{
-		EnsureLocalTableEmpty(relationId);
-	}
-	else if (distributionMethod != DISTRIBUTE_BY_HASH &&
-			 distributionMethod != DISTRIBUTE_BY_NONE)
-	{
-		EnsureLocalTableEmpty(relationId);
-	}
-	else if (!RegularTable(relationId))
+	/* we don't support copying local data via deprecated APIs */
+	bool shouldEnsureLocalTableEmpty = viaDepracatedAPI;
```

Comment:
I think in this method there are two things done:
- we have some conditions to check if the local table should be empty
- If it should be empty, we check if it is actually empty

So we can have a function that returns true or false based on the conditions, say `ShouldTableBeEmpty`. Then this method becomes pretty simple too:

```
if (ShouldTableBeEmpty(...) {
    EnsureLocalTableEmpty();
}
```

What do you think? Does it seem simpler for you?

Reply:
Actually this seem really simpler for me, I will apply that one too